### DISCUSSION TITLE
Fastlane Gym is complaining about env variable

### DISCUSSION THREAD

**Discussion by afa7789** [fastlane/fastlane#20385] [Q&A]
```
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.206.2/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in 
`user_error!': 'silent' value must be either `true` or `false`! Found String instead. 
(FastlaneCore::Interface::FastlaneError)
```

I am trying to run it this way: 
`fastlane gym --clean  --scheme "Marketplace" -allowProvisioningUpdates --verbose`
But I can't find how exactly to solve it.

I believe I'm getting this error because I tried to add the flag --silent on testing.

Tried to see where the environment is stored to check how to remove it, but it was unclear for me.

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 12.2.1                                                         |
| Ruby                        | 2.6.8                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.32.1 (Apple Git-133)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | macOS 12.2.1 (21D62)                                           |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 13.4.1                                                         |
| Swift Version               | 5.6.1                                                          |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | pt_BR.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Description of what the lane does"
  lane :custom_lane do
    # add actions here: https://docs.fastlane.tools/actions
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# app_identifier("[[APP_IDENTIFIER]]") # The bundle identifier of your app
# apple_id("[[APPLE_ID]]") # Your Apple email address


# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.206.2 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.3.0        |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.7.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.5.0        |
| CFPropertyList                  | 2.3.6        |
| public_suffix                   | 4.0.5        |
| addressable                     | 2.8.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| optparse                        | 0.1.1        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.3        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| excon                           | 0.76.0       |
| faraday                         | 1.0.1        |
| unf_ext                         | 0.0.7.7      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.3        |
| faraday-cookie_jar              | 0.0.6        |
| faraday_middleware              | 1.0.0        |
| fastimage                       | 2.2.0        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.1.0        |
| mini_magick                     | 4.10.1       |
| naturally                       | 2.2.0        |
| rubyzip                         | 2.3.0        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.0        |
| dotenv                          | 2.7.6        |
| bundler                         | 2.1.2        |
| simctl                          | 1.6.8        |
| jwt                             | 2.2.1        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| declarative-option              | 0.1.0        |
| representable                   | 3.0.4        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.0.2        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.1        |
| signet                          | 0.16.1       |
| googleauth                      | 1.1.3        |
| httpclient                      | 2.8.3        |
| webrick                         | 1.4.4        |
| google-apis-core                | 0.5.0        |
| google-apis-playcustomapp_v1    | 0.7.0        |
| google-apis-androidpublisher_v3 | 0.21.0       |
| google-cloud-env                | 1.3.3        |
| google-cloud-errors             | 1.0.1        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.10.0       |
| google-apis-storage_v1          | 0.14.0       |
| rake                            | 13.0.1       |
| digest-crc                      | 0.6.1        |
| google-cloud-storage            | 1.36.2       |
| emoji_regex                     | 3.0.0        |
| jmespath                        | 1.4.0        |
| aws-partitions                  | 1.354.0      |
| aws-eventstream                 | 1.1.0        |
| aws-sigv4                       | 1.2.1        |
| aws-sdk-core                    | 3.104.3      |
| aws-sdk-kms                     | 1.36.0       |
| aws-sdk-s3                      | 1.78.0       |
| rexml                           | 3.2.5        |
| nanaimo                         | 0.3.0        |
| colored2                        | 3.1.2        |
| claide                          | 1.0.3        |
| atomos                          | 0.1.3        |
| xcodeproj                       | 1.21.0       |
| forwardable                     | 1.2.0        |
| logger                          | 1.3.0        |
| stringio                        | 0.0.2        |
| ipaddr                          | 1.2.2        |
| openssl                         | 2.1.2        |
| ostruct                         | 0.1.0        |
| strscan                         | 1.0.0        |
| date                            | 2.0.0        |
| io-console                      | 0.4.7        |
| fileutils                       | 1.1.0        |
| etc                             | 1.0.1        |
| zlib                            | 1.0.0        |
| libxml-ruby                     | 3.2.1        |
| psych                           | 3.1.0        |
| mutex_m                         | 0.1.0        |
</details>

*generated on:* **2022-06-15**
</details>

**Comment by afa7789** [comment]
I tried to unset GYM_SILENT and FASTLANE_SILENT but both return an empty string on echo $GYM_SILENT and $FASTLANE_SILENT

**Comment by afa7789** [comment]
By running with silent it doesn't complain, is there such a way to help define it again as false ?

**Comment by rogerluan** [comment]
You can try printing your env variables on terminal by running the simple command `env`.
Alternatively, there may be things loading up more env vars during run time that are not present in your shell, so you could try creating a lane in your Fastfile to print your env like `puts ENV`. Try looking for env vars that are related to 'silent' or 'verbose', and see if they may be conflicting with the env vars that fastlane might be picking up

**Comment by afa7789** [comment]
```
TERM_PROGRAM=Apple_Terminal


SHELL=/bin/zsh

TERM=xterm-256color

TMPDIR=/var/folders/hp/90xgvr2d7114ctcxvwf8mzl80000gn/T/

TERM_PROGRAM_VERSION=443

TERM_SESSION_ID=D584119F-0F93-4E3D-AD29-7A10FF5B9CEB

USER=starwars

SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.ZGvjZkVBzH/Listeners

PATH=/usr/local/opt/php@8.0/sbin:/usr/local/opt/php@8.0/bin:/Users/starwars/opt/anaconda3/condabin:/usr/local/opt/php@7.4/sbin:/usr/local/opt/php@7.4/bin:/Users/starwars/.nvm/versions/node/v14.16.0/bin:~/.composer/vendor/bin:/usr/local/opt/libcouchbase@2/bin:/usr/local/opt/php@7.4/bin:/usr/local/sbin:/usr/local/opt/ruby/bin:/usr/local/opt/openssl@1.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Users/starwars/Library/Android/Sdk/tools:/Users/starwars/Library/Android/Sdk/platform-tools

LaunchInstanceID=88328BAC-CD45-45A0-B0E9-091F148F8116

__CFBundleIdentifier=com.apple.Terminal

PWD=/Users/starwars/projects/marketplace/react

XPC_FLAGS=0x0

XPC_SERVICE_NAME=0

SHLVL=1

HOME=/Users/starwars

LOGNAME=starwars

SECURITYSESSIONID=186a5

OLDPWD=/Users/starwars/automation/marketplace/react/marketplace-react-2

ZSH=/Users/starwars/.oh-my-zsh

ANDROID_HOME=/Users/starwars/Library/Android/Sdk

PAGER=less

LESS=-R

LSCOLORS=Gxfxcxdxbxegedabagacad

NVM_DIR=/Users/starwars/.nvm

NVM_CD_FLAGS=-q

NVM_BIN=/Users/starwars/.nvm/versions/node/v14.16.0/bin

NVM_INC=/Users/starwars/.nvm/versions/node/v14.16.0/include/node

CONDA_EXE=/Users/starwars/opt/anaconda3/bin/conda

_CE_M=

_CE_CONDA=

CONDA_PYTHON_EXE=/Users/starwars/opt/anaconda3/bin/python

CONDA_SHLVL=0

LANG=pt_BR.UTF-8

_=/usr/bin/env
```

doesnt look like so. Anyway, I'll keep testing with silent as true with --silent for now. Cause I gotta update at apple.
But I'm indeed curious about how could this happen.
